docs/docs/building-applications/5-async-queries.md

```diff
@@ -0,0 +1,70 @@
+---
+sidebar_position: 5
+---
+
+# Asynchronous Queries
+
+Groundlight provides a simple interface for submitting asynchronous queries. This is useful for times in which the thread or machine submitting image queries is not the same thread or machine that will be retrieving and using the results. For example, you might have a forward deployed robot or camera that submits image queries to Groundlight, and a separate server that retrieves the results and takes action based on them. We will refer to these two machines as the **submitting machine** and the **retrieving machine**.
+
+## Setup Submitting Machine
+On the **submitting machine**, you will need to install the Groundlight Python SDK. Then you can submit image queries asynchronously using the `ask_async` interface (read the full documentation [here](pathname:///python-sdk/api-reference-docs/#groundlight.client.Groundlight.ask_async)). `ask_async` submits your query and returns as soon as the query is submitted. It does not wait for an answer to be available prior to returning to minimize the time your program spends interacting with Groundlight. As a result, the `ImageQuery` object `ask_async` returns lacks a `result` (the `result` field will `None`). This is acceptable for this use case as the **submitting machine** is not interested in the result. Instead, the **submitting machine** just needs to communicate the `ImageQuery.id`s to the **retrieving machine** - this might be done via a database, a message queue, or some other mechanism. For this example, we assume you are using a database where you save the `ImageQuery.id` to it via `db.save(image_query.id)`.
+
+```python notest
+from groundlight import Groundlight
+import cv2
+from time import sleep
+
+detector = gl.get_or_create_detector(name="your_detector_name", query="your_query")
+
+cam = cv2.VideoCapture(0)  # Initialize camera (0 is the default index) 
+
+while True:  # TODO: add a way to exit this loop... not sure what makes sense here
```

Comment:
For reviewers - is `while True:` ok here? or is there some other reasonable way to communicate that you repeatedly query?

Reply:
Seems fine to me since you explain why we are iterating. You could also consider setting like an image query budget and iterate until the budge is not yet exceeded, so that there's a logical breaking condition. 

```python
target_num_image_queries = 100000
submitted_iqs = 0
while submitted_iqs < target_num_image_queries:
     pass
```

Reply:
I think `while True` is totally fine for sample code.  It's not uncommon in real code too.   You explicitly `break` when you want to.